packages/compiler/src/expression_parser/parser.ts

```diff
@@ -369,6 +381,16 @@ export class _ParseAST {
     this.index++;
   }
 
+  /**
+   * Executes a callback in the provided context.
+   */
+  private withContext<T extends(...args: any) => any>(context: ParseContext, cb: T): ReturnType<T> {
```

Comment:
This can be simplified to infer the return type directly in the call signature:

```suggestion
  private withContext<T>(context: ParseContext, cb: () => T): T {
```

Reply:
Nice, done.